src/lib.rs

```diff
@@ -433,12 +433,20 @@ impl<'input, Endian> Unit<'input, Endian>
 
             // Extract source file and line information about the compilation unit
             let line_offset = match entry.attr_value(gimli::DW_AT_stmt_list) {
-                Some(gimli::AttributeValue::DebugLineRef(offset)) => offset,
+                Ok(Some(gimli::AttributeValue::DebugLineRef(offset))) => offset,
+                Err(e) => {
+                    return Err(Error::from(ErrorKind::Gimli(e)))
+                        .chain_err(|| "invalid compilation unit statement list")
+                }
                 _ => return Ok(None),
             };
             let comp_dir = entry.attr(gimli::DW_AT_comp_dir)
+                .map_err(|e| Error::from(ErrorKind::Gimli(e)))
```

Comment:
None of the `map_err` cases are needed. They do change the error that is returned though. Something like:

    Error(Msg("test"), State { next_error: Some(Error(Gimli(BadUnsignedLeb128), State { next_error: None, backtrace: None })), backtrace: None }

vs

    Error(Msg("test"), State { next_error: Some(BadUnsignedLeb128), backtrace: None })

I'm not sure that the extra `Error(Gimli(..))` provides any value though, because it is behind a trait object pointer so you can't match on it. I haven't read much about matching on error chains though, so was hoping you had a reason for the `map_err` that I could learn from.